### PR TITLE
Association menus with restaurants, restrict menu creation to owners who have already set up their restaurant

### PR DESCRIPTION
**Pull request by AmberWilkie**
PT Story: https://www.pivotaltracker.com/story/show/131542173

Changes proposed in this pull request:
- Create belongs_to association restaurant to menu and has_many association menus to restaurant
- Restrict menu and dish creation to owners who have set up their restaurant
- Adjust almost all the cucumber tests to account for the new restriction on menu and dish creation

What I have learned working on this feature: Big takeaway from this feature for me is shifting from terminal to browser to allow two different systems to throw errors at me. Browser gave a much more helpful presentation of the same error (subsequently fixed, of course).

Ready for review @tochman !

### CODE REVIEW

File: app/controllers/dishes_controller.rb [hunk 1]
```diff
@@ -2,8 +2,13 @@ class DishesController < ApplicationController
   load_and_authorize_resource
 
   def new
-    @dish = Dish.new
-    @menus = Menu.all # Later on, we need to restrict this to only the menus of the current Owner
+    if current_user && current_user.role == 'owner' && !current_user.restaurant.nil?
```
Comment (tochman) [inline app/controllers/dishes_controller.rb]: This can be partially handled by CanCanCan

File: app/controllers/menus_controller.rb [hunk 2]
```diff
@@ -7,14 +7,20 @@ def index
   end
 
   def new
-    @menu = Menu.new
+    if current_user && current_user.role == 'owner' && !current_user.restaurant.nil?
```
Comment (tochman) [inline app/controllers/menus_controller.rb]: Same here. CanCanCan can be used for some of this functionality, like `current_user.role == 'owner'`
Also, we do have `current_user.owner?` method that could be used here.

File: app/controllers/menus_controller.rb [hunk 3]
```diff
@@ -7,14 +7,20 @@ def index
...
+    end
   end
 
   def show
   end
 
   def create
-    @menu = Menu.new(menu_params)
+    restaurant = Restaurant.find_by(user: current_user)
+    @menu = Menu.new(menu_params.merge({restaurant: restaurant}))
```
Comment (tochman) [inline app/controllers/menus_controller.rb]: Here we prefer to use `restaurant.menues.create(menu_params)`

**Review by tochman — CHANGES_REQUESTED** [review]
Can't do a full review tonight but made 3 comments so far.

File: app/controllers/dishes_controller.rb (hunk 1 above)
Comment (AmberWilkie) [reply app/controllers/dishes_controller.rb]: done

File: app/controllers/menus_controller.rb (hunk 2 above)
Comment (AmberWilkie) [reply app/controllers/menus_controller.rb]: done

File: app/controllers/menus_controller.rb (hunk 3 above)
Comment (AmberWilkie) [reply app/controllers/menus_controller.rb]: ok. Not stuck in my brain quite yet but getting there.

File: app/controllers/dishes_controller.rb [hunk 4]
```diff
@@ -2,8 +2,13 @@ class DishesController < ApplicationController
   load_and_authorize_resource
 
   def new
-    @dish = Dish.new
-    @menus = Menu.all # Later on, we need to restrict this to only the menus of the current Owner
+    if (can? :manage, Dish) && Restaurant.exists?(user: current_user)
```
Comment (tochman) [inline app/controllers/dishes_controller.rb]: I would like us to visit this line together. Not sure about `Restaurant.exists?(user: current_user)`

Comment (AmberWilkie) [reply app/controllers/dishes_controller.rb]: We discussed. It's now extracted to application controller.

**Review by tochman — APPROVED** [review]

**Pull request merged**